src/garage/torch/algos/sac.py

```diff
@@ -516,6 +516,17 @@ def to(self, device=None):
             device (str): ID of GPU or CPU.
 
         """
+
+        def optimizer_to(optim, device):
+            for param in optim.state.values():
```

Comment:
It would probably be better to have a seperate function (in `garage.torch._functions`) that recursively moves any nesting of dictionaries, lists, and tensors.

Reply:
Can optimizer state dicts be infinitely nested? I haven't really be able to find any documentation on it. Regardless, I feel like having a function that just moves optimizers might be easier to understand?

Reply:
Probably not optimizers themselves. 

Perhaps you could add this to our torch optimizer class, wherever we keep the special MAML optimizer.

Reply:
interestingly enough, the test would fail on the previous code anyways because it would crash due to things partially not being moved. I'm confused as to why changing the learning rate would have made it fail on the previous code if it ran, though.